src/Compilers/CSharp/Test/Semantic/Semantics/PatternSwitchTests.cs

```diff
@@ -1220,5 +1222,32 @@ public enum EnumA
 Default";
             var comp = CompileAndVerify(compilation, expectedOutput: expectedOutput);
         }
+
+        [Fact, WorkItem(10446, "https://github.com/dotnet/roslyn/issues/10446")]
+        public void InferenceInSwitch()
```

Comment:
>InferenceInSwitch [](start = 20, length = 17)

Consider adding a test for a scenario when pattern local is declared in ```when``` clause.
Consider adding a test for a scenario when ```i.ToString()``` is in ```when``` clause.